src/System.IO.Pipelines.Compression/CompressionPipelineExtensions.cs

```diff
@@ -124,7 +124,7 @@ public WritableDeflateTransform(CompressionLevel compressionLevel, int bits)
 
                     unsafe
                     {
-                        var handle = buffer.Pin();
+                        var handle = buffer.Retain(true);
```

Comment:
Specifically here only or in most/all places? What is your rationale for adding it here specifically?